pkg/generate/app/dockerimagelookup.go

```diff
@@ -48,31 +48,17 @@ func (r DockerClientSearcher) Search(terms ...string) (ComponentMatches, error)
 			return nil, err
 		}
 
-		glog.V(4).Infof("checking local Docker daemon for %q", ref.String())
-		images, err := r.Client.ListImages(docker.ListImagesOptions{})
-		if err != nil {
-			return nil, err
-		}
-
-		if len(ref.Registry) == 0 {
-			ref.Registry = "local Docker"
-		}
-		if len(ref.Tag) == 0 {
-			ref.Tag = imageapi.DefaultImageTag
-		}
-
 		termMatches := ScoredComponentMatches{}
-		for _, image := range images {
-			if tags := matchTag(image, term, ref.Registry, ref.Namespace, ref.Name, ref.Tag); len(tags) > 0 {
-				termMatches = append(termMatches, tags...)
-			}
-		}
-		sort.Sort(termMatches)
 
-		if r.RegistrySearcher != nil && len(termMatches.Exact()) == 0 {
+		// first look for the image in the remote docker registry
+		if r.RegistrySearcher != nil {
 			matches, err := r.RegistrySearcher.Search(term)
 			switch err.(type) {
 			case nil:
+				for k := range matches {
```

Comment:
i was originally thinking it was a key and not an index.  i'll change it, though honestly i don't see it as that important...